packages/central-server/src/apiV2/entityRelations/assertEntityRelationPermissions.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { TYPES } from '@tupaia/database';
+import { hasBESAdminAccess } from '../../permissions';
+import { mergeFilter, mergeMultiJoin } from '../utilities';
+
+export const assertEntityRelationPermissions = async (accessPolicy, models, entityRelationId) => {
+  const entityRelation = await models.entityRelation.findById(entityRelationId);
+  if (!entityRelation) {
+    throw new Error(`No entity relation exists with id ${entityRelationId}`);
+  }
+
+  const child = await models.entity.findById(entityRelation.child_id);
+  if (!accessPolicy.allows(child.country_code)) {
+    throw new Error('You do not have permissions for this entity relation');
+  }
+
+  return true;
+};
+
+export const createEntityRelationDbFilter = (accessPolicy, criteria, options) => {
+  const dbConditions = { ...criteria };
+  const dbOptions = { ...options };
+
+  if (!hasBESAdminAccess(accessPolicy)) {
+    // Our permissions check logic is simplified based on the following assumptions:
+    // 1. Each child is associated to a non-empty subset of its parent's countries
+    // 2. Projects cannot be children of other entities
+    //
+    // All valid relations should satisfy the above invariants.
+    // Note that invalid relations can expose parents that the user doesn't have permissions to,
+    // e.g. entityInCountryA -> countryB, where the user only has permissions to countryB
+    dbConditions['child.country_code'] = mergeFilter(
```

Comment:
I don't think that `mergeFilter` is really required here,, since

* There is no `child` table in the DB currently
* Users can filter based on columns from other tables, but I don't think they can use aliases for them

However, it's cheap to use and it will be more robust in case any of the above changes in the future